src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -69,6 +69,25 @@ class WorkflowManagerActor(backend: Backend) extends Actor with CromwellActor {
 
   private val workflowStore = new WriteOnceStore[WorkflowId, WorkflowActorRef]
 
+  if (getAbortJobsOnTerminate) {
+    Runtime.getRuntime.addShutdownHook(new Thread() {
+      override def run(): Unit = {
+        log.info(s"$tag: Received shutdown signal. Aborting all running workflows...")
+        workflowStore.toMap.foreach{case (id, actor)=>
+          CromwellServer.workflowManagerActor ! WorkflowManagerActor.WorkflowAbort(id)
+        }
+        var numRemaining = -1
+        while(numRemaining != 0) {
+          Thread.sleep(1000)
+          val result = globalDataAccess.getWorkflowsByState(Seq(WorkflowRunning, WorkflowAborting))
+          numRemaining = Await.result(result,Duration.Inf).size
```

Comment:
There's another val here, and an Await.result - which have caused us horrible problems in the past (I think @mcovarr is undergoing some pain right now to remove a few more).

If you were to do this more like SingleWorkflowRunnerActor, it would be a FSM with the number of outstanding child actors stored as part of the state. That's quite a big change just to modify the shutdown behaviour though. 

Anyone have any better ideas?